substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -0,0 +1,246 @@
+            ImageSingletons.add(SerializationRegistry.class, serializationSupport);
+        }
+
+        ReflectionDataBuilder reflectionData;
+        if (ImageSingletons.contains(RuntimeReflectionSupport.class)) {
+            reflectionData = (ReflectionDataBuilder) ImageSingletons.lookup(RuntimeReflectionSupport.class);
+        } else {
+            reflectionData = new ReflectionDataBuilder(access);
+            ImageSingletons.add(RuntimeReflectionSupport.class, reflectionData);
+        }
+
+        SerializationParserFunction serializationAdapter = (strTargetSerializationClass, checksum) -> {
+            Class<?> serializationTargetClass = resolveClass(strTargetSerializationClass, imageClassLoader);
+            UserError.guarantee(serializationTargetClass != null, "Cannot find serialization target class %s. The missing of this class can't be ignored even if -H:+AllowIncompleteClasspath is set." +
+                            " Please make sure it is in the classpath", strTargetSerializationClass);
+            if (Serializable.class.isAssignableFrom(serializationTargetClass)) {
+                Class<?> targetConstructor = serializationSupport.addSerializationConstructorAccessorClass(serializationTargetClass, checksum);
+                addReflections(reflectionData, serializationTargetClass, targetConstructor);
+            }
+        };
+
+        SerializationConfigurationParser parser = new SerializationConfigurationParser(serializationAdapter);
+        loadedConfigurations = ConfigurationParserUtils.parseAndRegisterConfigurations(parser, imageClassLoader, "serialization",
+                        ConfigurationFiles.Options.SerializationConfigurationFiles, ConfigurationFiles.Options.SerializationConfigurationResources,
+                        ConfigurationFiles.SERIALIZATION_NAME);
+        serializationSupport.reportErrors(access);
+    }
+
+    public static void addReflections(ReflectionDataBuilder reflectionData, Class<?> serializationTargetClass, Class<?> targetConstructor) {
+        if (targetConstructor != null) {
+            try {
+                reflectionData.register(targetConstructor.getDeclaredConstructor());
+            } catch (NoSuchMethodException e) {
+                VMError.shouldNotReachHere();
+            }
+        }
+
+        if (Externalizable.class.isAssignableFrom(serializationTargetClass)) {
+            try {
+                reflectionData.register(serializationTargetClass.getDeclaredConstructor((Class<?>[]) null));
+            } catch (NoSuchMethodException e) {
+                VMError.shouldNotReachHere();
+            }
+        }
+
+        reflectionData.register(serializationTargetClass);
+        boolean registerAllMethods = false;
+        boolean registerAllFieds = false;
+        try {
+            serializationTargetClass.getDeclaredField("serialVersionUID");
+        } catch (NoSuchFieldException e) {
+            registerAllMethods = true;
+            registerAllFieds = true;
```

Comment:
Why is field/method access not required when `serialVersionUID` is present?

Reply:
`getDeclaredMethods`, `getDeclaredFields` and `getDeclaredConstructors` are only called by method `computeDefaultSUID` when `serialVersionUID` is not present.